### PR TITLE
Cards Download in Appropriate Language

### LINKED ISSUE #841: allow to define a language for card picture downloads
A brazilian guy told me this... he said that this would be a very huge improvement since many people struggle with english in general (true for other languages/countries as well!). If they get the card pictures and/or text in the language of their choice that would help many foreign players and strengthens the international cockatrice community in general.
Everybody sees the pictures and/or text in their preferred languages, even in the same game. For many users this would be much more comfortable.

I found out after a check that magiccards.info and gatherer both provide card pictures (and -text) in different languages!

**Gatherer** (you need to click on languages to see all available (printed) languages)
Quiet Contemplation (en)
Quiet Contemplation (de)
Quiet Contemplation (chinese simplified)
As you see the multiverse id is different for each language, even for the same card.

**magiccards.info**
http://magiccards.info/ktk/en/50.html
http://magiccards.info/ktk/de/50.html
http://magiccards.info/ktk/cn/50.html
The direct link to the images are "magiccards.info/scans/**en**/ktk/50.jpg" etc.

So in theory this should be possible...

---

**Edit:**
In the meantime, mci redirects all requests to **Scryfall**, which has different arts, too as [it's part of scryfall now](https://scryfall.com/blog/scryfall-magiccards-info-190)!
So the above mci links point to this scryfall pages:
https://scryfall.com/card/ktk/50/en/
https://scryfall.com/card/ktk/50/de/
https://scryfall.com/card/ktk/50/zhs/

### PR DESCRIPTION
**Pull request by marcofernandezheras**
Fix #841

### CODE REVIEW

File: cockatrice/src/carddatabase.cpp [hunk 1]
```diff
@@ -340,6 +340,71 @@ void PictureLoader::processLoadQueue()
...
+        langDict["es"] =   "Español";
+        langDict["fr"] =   "Français";
+        langDict["it"] =   "Italiano";
+        langDict["ja"] =   "日本語";
+        langDict["ko"] =   "한국어";
+        langDict["pt_BR"] =  "Português";
+        langDict["pt"] =     "Português";
+        langDict["ru"] =     "русский язык";
+        langDict["zh-Hans"] = "简体中文";
+    }
```
Comment (tooomm) [inline cockatrice/src/carddatabase.cpp]: where do you take this languages from?
are they gatherer related? are they related to our language settings or files in cockatrice?

Comment (marcofernandezheras) [inline cockatrice/src/carddatabase.cpp]: the Map key is language settings related ( returned by settingsCache->getLang() ) and the values are gatherer related, used to search for muid translation (it the "Translated Language" column of the gathere language page)

File: cockatrice/src/carddatabase.cpp [hunk 2]
```diff
@@ -340,6 +340,71 @@ void PictureLoader::processLoadQueue()
...
 
+int PictureLoader::getIdFromHtml(QString html, QString lang)
+{
+    QStringList langs = html.split("class=\"cardItem");
+    for(int i = 1; i < langs.size(); ++i)
+    {
+        const QString langHtml = langs.at(i);
+        if(langHtml.contains(lang,Qt::CaseInsensitive))
+        {
+            //qDebug() << langHtml;
```
Comment (ZeldaZach) [inline cockatrice/src/carddatabase.cpp]: This line should be removed

File: cockatrice/src/carddatabase.cpp [hunk 3]
```diff
@@ -16,7 +16,7 @@
 #include <QDebug>
 #include <QImageReader>
 #include <QMessageBox>
-
+#include <QEventLoop>
```
Comment (ZeldaZach) [inline cockatrice/src/carddatabase.cpp]: You should keep a line separator between the includes and the code to the file

**Comment by ZeldaZach** [comment]
My initial review shows that there are a lot of spacing issues. Try to use only one space on either side of the `=` symbol(s). Also, you should really do this on a sub-branch on your fork instead of the master.

**Comment by marcofernandezheras** [comment]
Moved to a new branch, making PR from there

File: cockatrice/src/carddatabase.cpp (hunk 1 above)
Comment (tooomm) [inline cockatrice/src/carddatabase.cpp]: Ok. Thanks for the insight!

2 things:
- cs is Česky (Czech) ;)<br>
I guess you use it for Chinese Traditional? The correct (not yet in the client, but you should implement+support it already) is "**zh-Hant**"
- You let the client download Portuguese cards for people that use either Portuguese or Brazilian Portuguese as their client language? That should be fine, I talked to two Brazilians.

You might want to update that in your new PR. ;)

**Pull request closed without merging**